### PR TITLE
2015 01 joka refactore for better documentation and clear modules responsibility

### PR DESCRIPTION
**Pull request by joka**
* support readthedocs.org: http://adhocracy3mercator.readthedocs.org/

* document backend modules api:
 failed, only docstring are visible (strange sphinx behavior, if i recreate all documentation it works) 
  http://adhocracy3mercator.readthedocs.org/en/latest/api_backend/adhocracy_core/adhocracy_core.authentication.html#module-adhocracy_core.authentication

* check that document generation runs without warnings: 
  failed, (strange sphinx behavior)

* document  backend modules responsibilities and dependencies:
   http://adhocracy3mercator.readthedocs.org/en/latest/style_guide/backend_structure.html#modules-api-and-separation-of-responsibility

* refactor code regarding modules responsibilites: mostly done

NOTE: due to rebasing there are some commits from Christion in the commit list below, just ignore them.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,3 @@
-sudo: false
```
Comment (ChristianSi) [inline .travis.yml]: Why is the sudo: false gone?

File: README.rst [hunk 2]
```diff
@@ -3,9 +3,11 @@ Adhocracy customizations for advocate-europe
 
 .. image:: https://api.travis-ci.org/liqd/adhocracy3.mercator.png?branch=master
     :target: http://travis-ci.org/liqd/adhocracy3.mercator
-.. image:: https://coveralls.io/repos/liqd/adhocracy3.mercator/badge.png?branch=master
+.. image:: https://coveralls.io/repos/liqd/adhocracy3.mercatorz/badge.png?branch=master
```
Comment (ChristianSi) [inline README.rst]: That's a regression too. Looks like some changes by been reverted by rebasing. I'm afraid I cannot merge this as is :(

File: docs/glossary.rst [hunk 3]
```diff
@@ -32,24 +32,23 @@ Glossary
        For more information about the permission system read
        :doc:`api/authentication_api`.
 
-   userid:
+   userid
       The unique id for one userique id of one :term:`group`: "group:<name>".
 
    group
-       A set of users. Can be mapped to permission :term:`role`s.
+       A set of users. Can be mapped to permision :term:`role`s.
```
Comment (ChristianSi) [inline docs/glossary.rst]: This looks like a regression too. Not sure about same of the other changes in the file...

File: src/adhocracy_core/adhocracy_core/interfaces.py [hunk 4]
```diff
@@ -141,6 +141,16 @@ def get(params: dict={}) -> dict:
         of the returned dictionary, e.g. query parameters in a GET request
         """
 
+    def get_cstruct(request, params: dict={}):
+        """Return cstruct data.
+
+        Bind `request` and `context` to colander schema.
```
Comment (ChristianSi) [inline src/adhocracy_core/adhocracy_core/interfaces.py]: `context` is mentioned in the docstring but there is no such attribute?

File: src/adhocracy_core/adhocracy_core/sheets/__init__.py [hunk 5]
```diff
@@ -134,6 +148,10 @@ def _get_back_reference_appstruct(self, params: dict={}) -> dict:
                 yield(key, node_backrefs)
 
     def _get_backrefs(self, node: Reference) -> dict:
+        """Return backreference data.
+
+        This might be overridden in subclasses.
+        """""
```
Comment (ChristianSi) [inline src/adhocracy_core/adhocracy_core/sheets/__init__.py]: Too many closing quotes.

File: src/adhocracy_core/adhocracy_core/sheets/__init__.py [hunk 6]
```diff
@@ -217,10 +235,40 @@ def _notify_resource_sheet_modified(self,
                                           request)
             registry.notify(event)
 
+    def get_cstruct(self, request: Request, params: dict={}):
+        """Return cstruct data.
+
+        Bind `request` and `context` to colander schema.
```
Comment (ChristianSi) [inline src/adhocracy_core/adhocracy_core/sheets/__init__.py]: Not quite clear where `context` comes from.

File: src/adhocracy_core/adhocracy_core/sheets/test_asset.py [hunk 7]
```diff
@@ -245,6 +247,8 @@ def test_crop_if_needed_no_cropping_needed_due_to_rounding(
 
 
 class TestRetrieveAssetFile:
+     # TODO Why do we need this test? If a sheet is not registered something
+     # is really broken and we get and internal error anyway.
```
Comment (ChristianSi) [inline src/adhocracy_core/adhocracy_core/sheets/test_asset.py]: I suspect we need it to get 100% test coverage :)

File: src/adhocracy_core/versions.cfg [hunk 8]
```diff
@@ -50,8 +49,9 @@ repoze.sendmail = 4.2
 requests = 2.5.1
 rubygemsrecipe = 0.2.1
 selenium = 2.44.0
-sphinxcontrib-napoleon = 0.2.9
-sphinxcontrib-zopeext = 0.2.1
+Babel = 1.3
+repoze.sphinx.autointerface = 0.7.1
+snowballstemmer = 1.2.0
```
Comment (ChristianSi) [inline src/adhocracy_core/versions.cfg]: Interesting, where do we need that one? (Just curious.)

**Comment by ChristianSi** [comment]
Looks very good, except for the few things i noted.

Tests pass for me.

@joka Feel free to merge after addressing my comments.

**Comment by ChristianSi** [comment]
One test failed in Travis (TestMercatorForm.test_fill_all_fields), but I guess that may be random noise (as long as we still have the old spliter tests).

File: src/adhocracy_core/versions.cfg (hunk 8 above)
Comment (joka) [inline src/adhocracy_core/versions.cfg]: > Interesting, where do we need that one? (Just curious.)
Babel/snowballstemmer  is just a Sphinx dependency.
spincontrib-napolen is included in Spinx > =3
sphinxcontrib-zopeext is replaced by repoze.sphinx.autointerface

**Pull request merged**